presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -748,6 +748,7 @@ private void startInBackground(ManagedQueryExecution query)
         synchronized (root) {
             runningQueries.add(query);
             InternalResourceGroup group = this;
+            query.setResourceGroupQueryLimits(group.getResourceGroupQueryLimits());
```

Comment:
lets move this to line 688 - before - `if (canRun && queuedQueries.isEmpty()) {` - if I had said something else earlier - lets forget it ;) 

The reason is that `startInBackground` maybe invoked at some random time down the stack and it might be too late. By invoking it earlier - we will set the value as soon as the query is submitted to the resource group which should happen in the relative earlier stages of the query.